src/seedu/addressbook/data/person/Address.java

```diff
@@ -38,22 +54,128 @@ public static boolean isValidAddress(String test) {
 
     @Override
     public String toString() {
-        return value;
+        String address = "";
+        address += block + SPLIT_TOKEN;
+        address += street + SPLIT_TOKEN;
+        address += unit + SPLIT_TOKEN;
+        address += postalCode;
+
+        return address;
     }
 
     @Override
     public boolean equals(Object other) {
         return other == this // short circuit if same object
                 || (other instanceof Address // instanceof handles nulls
-                && this.value.equals(((Address) other).value)); // state check
+                && this.toString().equals(((Address) other).toString())); // state check
     }
 
     @Override
     public int hashCode() {
-        return value.hashCode();
+        return this.toString().hashCode();
     }
 
     public boolean isPrivate() {
         return isPrivate;
     }
 }
+
+class Block {
```

Comment:
This class and its non-trivial methods should have header comments. 
If not, developers who write client code for this class will not know how to use it.